test/e2e_node/node_problem_detector_linux.go

```diff
@@ -207,7 +239,7 @@ var _ = framework.KubeDescribe("NodeProblemDetector [NodeFeature:NodeProblemDete
 						{
 							Name:    name,
 							Image:   image,
-							Command: []string{"sh", "-c", "touch " + logFile + " && /node-problem-detector --logtostderr --system-log-monitors=" + configFile + fmt.Sprintf(" --apiserver-override=%s?inClusterConfig=false", framework.TestContext.Host)},
+							Command: []string{"sh", "-c", "touch " + logFile + " && /node-problem-detector --logtostderr --system-log-monitors=" + configFile + fmt.Sprintf(" --apiserver-override=%s?inClusterConfig=false&useServiceAccount=true&insecure=true", framework.TestContext.Host)},
```

Comment:
Why `insecure=true`? I guess `insecure` here means not validating server cert instead of using `http`?

Reply:
> Why `insecure=true`?

Perhaps I am missing something, I think we don't know the CA used by apiserver in node e2e tests.

> I guess insecure here means not validating server cert instead of using http?

nope, it is equivalent to `kubectl --insecure-skip-tls-verify ...`: https://github.com/kubernetes/node-problem-detector/blob/53ee78d3725f157f12fd19978b94b2fe8978150b/vendor/k8s.io/heapster/common/kubernetes/configs.go#L54
we are still using HTTPS.

Reply:
looks like we could use `inClusterConfig`, so that we dont need `insecure=true` here.